ponzu/llama/curatedTVL.py

```diff
@@ -0,0 +1,130 @@
+from llama.llama import Llama
+import pandas as pd
+
+class CuratedTVL(Llama):
+    def __init__(self):
+        super().__init__()  # call Llama's __init__ method
+        self.tvl = None
+        self.raw_tvl = None
+        self.curated_tvl = None
+
+    def fetch_tvl(self, chains):
+        if self.raw_tvl is None:
+            self.raw_tvl = self.getChainTVL(chains)
+            self._reclassify_tvl()
+        return self.raw_tvl
+
+    def _reclassify_tvl(self):
+        self.raw_tvl.loc[self.raw_tvl['protocol'] == 'eos-rex', 'category'] = 'Liquid Staking'
+        self.raw_tvl.loc[self.raw_tvl['protocol'] == 'atomichub', 'category'] = 'NFT Marketplace'
+
+    def curate_tvl(self, chains):
+        curated_chains = ['ethereum','tron','binance','arbitrum','polygon','optimism','avalanche','fantom','solana','cardano','eos','tezos','aptos','near','stacks','sui','hedera','aurora','eos evm','wax','europa']
+        for chain in chains:
+            if chain not in curated_chains:
+                return f"Error: {chain} is not in the curated list of chains." 
+        if self.raw_tvl is None:
+            self.fetch_tvl(chains)
+
+        self._filter_tvl()
+        return self.curated_tvl
+
+    def _filter_tvl(self):
+        df = self.raw_tvl[(self.raw_tvl['type'] == 'tvl') | (self.raw_tvl['type'] == 'borrowed')]
+        exclude_categories = ['CEX', 'Chain', 'Yield', 'Bridge', 'Liquid Staking', 'Liquidity manager', 'RWA', 'Gaming', 'NFT Marketplace', 'Leveraged Farming', 'Prediction Market']
+        df = df[~df['category'].isin(exclude_categories)]
+        exclude_protocols = ['ribbon','frax','unslashed','sturdy','alongside','illuminate','instadapp','defisaver','pooltogether','cian','ubiquity-dao','sandclock','templedao','unbound','axia-protocol','tornado-cash','aztec','tronnrg','garble.money','sperax-demeter','volta-finance','uniwswap-unia-farms','mm-stableswap-polygon','zkbob','brokkr-finance','defrost','sherpa-cash','sturdy','shadecash','zeta']
```

Comment:
Can you pull the 'exclude protocols' and 'curated chains' and the like out into a config file? that way its a lot more clear what we are doing and easier to edit later

Reply:
yes